src/scheduled_tasks.rs

```diff
@@ -33,6 +41,11 @@ pub fn setup(db_url: String) -> Result<(), LemmyError> {
     clear_old_activities(&mut conn);
   });
 
+  update_instance_software(&mut conn_2, &client);
+  scheduler.every(1.days()).run(move || {
+    update_instance_software(&mut conn_2, &client);
```

Comment:
Runs on startup, and every day. Seems like a daily run is best. The software is never gonna change, but the software version might.